### PR TITLE
bug: Sign message with ethereum signed prefix

### PR DESCRIPTION
**Pull request by oneleo**
## Description

- Note: GitHub Action will only pass after https://github.com/pilagod/waallet-contract/pull/39 PR for the wAAllet contract is merged

### CODE REVIEW

File: packages/account/PasskeyAccount/passkeyOwnerP256.test.ts [hunk 1]
```diff
@@ -1,10 +1,12 @@
+import { toUtf8Bytes } from "ethers"
+
 import { PasskeyOwnerP256 } from "./passkeyOwnerP256"
 
 describe("PasskeyOwnerP256", () => {
   const owner = new PasskeyOwnerP256()
 
   it("should sign challenge", async () => {
-    const challenge = "challenge"
+    const challenge = toUtf8Bytes("challenge")
```
Comment (pilagod) [inline packages/account/PasskeyAccount/passkeyOwnerP256.test.ts:9]: Q: Why it didn't need this conversion before?

File: packages/account/PasskeyAccount/passkeyOwnerP256.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { p256 } from "@noble/curves/p256"
 import * as ethers from "ethers"
 
 import byte from "~packages/util/byte"
+import cryptography from "~packages/util/cryptography"
```
Comment (pilagod) [inline packages/account/PasskeyAccount/passkeyOwnerP256.ts:5]: I think we don't need additional wrapper for it, we can directly use `ethers.hashMessage`.

File: packages/account/PasskeyAccount/passkeyOwnerP256.test.ts (hunk 1 above)
Comment (oneleo) [reply packages/account/PasskeyAccount/passkeyOwnerP256.test.ts:9]: The 'challenge' fails ethers.isBytesLike().
However, toUtf8Bytes('challenge') is no longer used after removing the cryptography util.

File: packages/account/PasskeyAccount/passkeyOwnerP256.test.ts [hunk 3]
```diff
@@ -8,4 +14,42 @@ describe("PasskeyOwnerP256", () => {
     const signature = await owner.sign(challenge)
     console.log(signature)
   })
+
+  it("should verify signature", async () => {
+    const challenge = "challenge"
+    const signature = await owner.sign(challenge)
+    const signatureDecoded = AbiCoder.defaultAbiCoder().decode(
```
Comment (pilagod) [inline packages/account/PasskeyAccount/passkeyOwnerP256.test.ts:21]: nit: Maybe directly give them a name for readability. 

```suggestion
    const [, authenticatorData,, clientDataJSON,,, r, s] = AbiCoder.defaultAbiCoder().decode(
```

Comment (pilagod) [reply packages/account/PasskeyAccount/passkeyOwnerP256.test.ts:21]: And in the future, the passkey owner should be refactored into more general shape, which will not be bound to passkey account data structure.

**Review by pilagod — APPROVED** [review]
Before merging this PR, we should first merge its counterpart in the contract repo, and rerun CI to check this PR meets our expectation.

**Pull request merged**